controllers/hvpa_controller.go

```diff
@@ -274,14 +274,22 @@ func (r *HvpaReconciler) deleteHvpaFinalizers(hvpa *autoscalingv1alpha1.Hvpa) {
 	}
 }
 
-func getVpaFromHvpa(hvpa *autoscalingv1alpha1.Hvpa) *vpa_api.VerticalPodAutoscaler {
+func getVpaFromHvpa(hvpa *autoscalingv1alpha1.Hvpa) (*vpa_api.VerticalPodAutoscaler, error) {
+	labels := hvpa.Spec.Vpa.Template.GetLabels()
+
+	if labels == nil || len(labels) == 0 {
+		// TODO: Could be done better as part of validation
+		return nil, fmt.Errorf("Need labels in VPA template")
+	}
+
 	// Updater policy set to "Off", as we don't want vpa-updater to act on recommendations
 	updatePolicy := vpa_api.UpdateModeOff
 
 	return &vpa_api.VerticalPodAutoscaler{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      hvpa.Name + "-vpa",
 			Namespace: hvpa.Namespace,
+			Labels:    labels,
```

Comment:
What about annotations? Why not copy `hvpa.Spec.Vpa.Template.ObjectMeta` and override only `Name` and `Namespace`?

Reply:
Fixed [de8e35e](https://github.com/gardener/hvpa-controller/pull/26/commits/de8e35e64ae86f048cc812d02c33853fc46fb16e)